### PR TITLE
Tutorial2

### PR DESCRIPTION
**Pull request by auchers**
tutorial2_barChart

It's tricky to decide what to include. 
This is what I settled on for now, but definitely open to other thoughts. 

**This week**
- [x] bars - `rects`, `text`, all in a `group` that we move with `transform`s
- [x] button control - for re-sort on click
- [x] `enter` and `update` lifecycle stages - requires data join with a `key`
- [x] `transition` with `delay`
- [x] CSS pseudo `:hover` selector for mouse interaction
Note: data is a randomly generated array of numbers.

**Thoughts on assignment/bonus**
- [ ] something with color scheme (implementing their own scale)
??? - I'll keep thinking about it
TODO: need to write up this week's `README`

**Save for next week**  
- [ ] Axes
- [ ] `Exit` selection
- [ ] more on data accessors (dealing with json, rather than simple array)

### CODE REVIEW

**Comment by auchers** [comment]
Feb-04-2020 23-34-07

refactor to 
- [x] remove draw function
- [x] remove enter-update
- [x] add axis

Note: Orange border is meant for illustration purposes in the live tutorial -- we can remove this. Also can restyle the whole thing -- did I hear #steelblue?

File: tutorial2_quantities_and_amounts/demo/style.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+.hover-text {
+  text-anchor: middle;
+  fill: white;
+  opacity: 0;
+}
+
+/* when the user hovers over a bar, show the text */
+g:hover .hover-text {
+  opacity: 1;
+}
```
Comment (efrymire) [inline tutorial2_quantities_and_amounts/demo/style.css]: this didn't seem to work for me -- looks like only when I click the bar it shows. I also don't know if we would have time in class / this week. How important to you is it? We could also just cover it, but not include it in the tutorial

File: tutorial2_quantities_and_amounts/demo/main.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  };
+}).then(data => {
+  console.log(data);
+  /** CONSTANTS */
+  const width = window.innerWidth * 0.9,
+    height = window.innerHeight / 3,
+    paddingInner = 0.2,
+    margin = { top: 20, bottom: 20, left: 50, right: 50 },
+    duration = 1000,
+    delay = 20;
```
Comment (efrymire) [inline tutorial2_quantities_and_amounts/demo/main.js]: this is good coding practice, but wondering if keeping them up here is disjointed from what is happening below. I think it makes sense for all css related items (`width`, `height`, `paddingInner`, `margin`) but maybe, just to keep it as simple as possible and this is a low hanging fruit, we could move `duration` and `delay` values to the code? This isn't a blocker but was just thinking -- trying to make the code below as obvious as possible (for example, someone who didn't notice we wrote this at the top might think `duration` is a JS defined item or a d3.js term)

**Review by efrymire — CHANGES_REQUESTED** [review]

File: tutorial2_quantities_and_amounts/demo/main.js (hunk 2 above)
Comment (auchers) [reply tutorial2_quantities_and_amounts/demo/main.js]: yea that's fair -- I was planning on having a slide in my deck talking about how we are going to start organizing our code and to specifically call out the idea of defining all the constants at the top. Something like an outline for our files going on out.

we can definitely pair it down though - whatever you think is best!

**Review by efrymire — APPROVED** [review]

**Pull request merged**